pwndbg/lib/config.py

```diff
@@ -2,16 +2,41 @@
 from functools import total_ordering
 from typing import List
 
+import gdb
+
+PARAM_CLASSES = {
+    # The Python boolean values, True and False are the only valid values.
+    bool: gdb.PARAM_BOOLEAN,
+    # This is like PARAM_INTEGER, except 0 is interpreted as itself.
+    int: gdb.PARAM_ZINTEGER,
+    # When the user modifies the string, any escape sequences,
+    # such as ‘\t’, ‘\f’, and octal escapes, are translated into
+    # corresponding characters and encoded into the current host charset.
+    str: gdb.PARAM_STRING,
+}
+
 
 # @total_ordering allows us to implement `__eq__` and `__lt__` and have all the
 # other comparison operators handled for us
 @total_ordering
 class Parameter:
-    def __init__(self, name, default, docstring, scope="config"):
+    def __init__(
+        self,
+        name,
+        default,
+        docstring,
+        help_docstring=None,
+        param_class=None,
+        enum_sequence=None,
+        scope="config",
+    ):
```

Comment:
Maybe we should also add the `*` in here to force keyword params starting from `help_docstring`?

Reply:
That probably makes sense, :+1:.

Btw one note: `docstring` and `help_docstring` is not so obvious I guess? I already merged this but noticed this only now. Maybe we should rename `docstring` to `set_show_doc`?